RasterPropMonitor/Core/VariableOrNumber.cs

```diff
@@ -54,6 +52,9 @@ public class VariableOrNumber
 
         public bool isConstant => updateType == VariableUpdateType.Constant;
 
+        private double numericValue;
+        private string stringValue;
```

Comment:
So it turns out that volatile variables do not update these cached values, so reading them directly is a bug.